### PR TITLE
Use Cork for output

### PR DESCRIPTION
**Pull request by DanToml**
fixes https://github.com/danger/danger/issues/132

- [x] Use Cork in Dangerfile
- [x] Pass Cork into Dangerfile from runner
- [x] Use Cork in Local
- [x] Use Cork in messaging
- [x] Use Cork in new_plugin
- [x] Use Cork in Interviewer

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Master
 
+* Use Cork for consolore output. - DanToml
```
Comment (segiddins) [inline CHANGELOG.md]: `Consolore` ?

File: lib/danger/commands/runner.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@ def initialize(argv)
       super
     end
 
+    def ui
+      @board ||= Cork::Board.new
```
Comment (segiddins) [inline lib/danger/commands/runner.rb]: should pass `--verbose` and `--silent` to this

File: lib/danger/danger_core/dangerfile.rb [hunk 3]
```diff
@@ -188,29 +189,29 @@ def print_results
...
+          params[:title] = case current
+                           when :errors
+                             current.to_s.capitalize.red
+                           when :warnings
+                             current.to_s.capitalize.yellow
+                           else
+                             current.to_s.capitalize
+                           end
+
+          ui.puts ""
```
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: the empty string here isn't necessary

File: lib/danger/danger_core/dangerfile.rb [hunk 4]
```diff
@@ -188,29 +189,29 @@ def print_results
...
+          ui.puts Terminal::Table.new(params)
+          ui.puts ""
+        end
 
-      puts "Markdown: ".green if status[:markdowns].count > 0
-      status[:markdowns].each do |current_markdown|
-        puts current_markdown
+        ui.puts "Markdown: ".green if status[:markdowns].count > 0
```
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: should be a sectio

**Comment by DanToml** [comment]
There are still a couple of places using `puts`/`print`, but that's in `Interviewer`, and an error dump in `GitHub`, and I'm not sure of a good way to handle those.

**Comment by DanToml** [comment]
screen shot 2016-06-09 at 21 45 49

**Comment by segiddins** [comment]
@DanToml try injecting a board?

File: lib/danger/danger_core/dangerfile.rb [hunk 5]
```diff
@@ -159,58 +160,60 @@ def parse(path, contents = nil)
...
+          next unless params[:rows].count > 0
+          params[:title] = case current
+                           when :errors
+                             current.to_s.capitalize.red
+                           when :warnings
+                             current.to_s.capitalize.yellow
+                           else
+                             current.to_s.capitalize
+                           end
+          ui.puts Terminal::Table.new(params)
```
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: no, the puts are necessary -- they add blank lines before & after the table. all that was unnecessary was the argument

Comment (DanToml) [inline lib/danger/danger_core/dangerfile.rb]: so just `ui.puts`?

**Comment by DanToml** [comment]
screen shot 2016-06-09 at 22 21 57

File: lib/danger/danger_core/dangerfile.rb (hunk 5 above)
Comment (segiddins) [inline lib/danger/danger_core/dangerfile.rb]: yup

**Comment by DanToml** [comment]
Now injects a board into `Interviewer`

**Comment by orta** [comment]
Awesome, looks good from my side -anyone want to give a 2nd opinion and merge?

**Comment by segiddins** [comment]
My only thing would be to ask that all output be redirected during the tests now that we have a way to do that

**Pull request merged**